tensorboard/plugins/histogram/histograms_plugin.py

```diff
@@ -51,16 +68,62 @@ def is_active(self):
     return bool(self._multiplexer) and any(self.index_impl().values())
 
   def index_impl(self):
-    return {
-        run_name: run_data[event_accumulator.HISTOGRAMS]
-        for (run_name, run_data) in self._multiplexer.Runs().items()
-        if event_accumulator.HISTOGRAMS in run_data
-    }
+    """Return {runName: {tagName: {displayName: ..., description: ...}}}."""
+    runs = self._multiplexer.Runs()
+    result = {run: {} for run in runs}
+
+    # Old-style runs
+    for (run, run_data) in six.iteritems(runs):
+      for tag in run_data.get(event_accumulator.HISTOGRAMS, []):
+        result[run][tag] = {'displayName': tag, 'description': ''}
+
+    # New-style runs
+    mapping = self._multiplexer.PluginRunToTagToContent(metadata.PLUGIN_NAME)
+    for (run, tag_to_content) in six.iteritems(mapping):
+      for (tag, content) in six.iteritems(tag_to_content):
+        content = metadata.parse_summary_metadata(content)
```

Comment:
Just pointing out that when we change to 1st class descr/display name, we'll need to continue looking for it here in histograms, for backcompat. Add a comment?